hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -119,8 +119,10 @@ private[hudi] object HoodieSparkSqlWriter {
 
       // Create the table if not present
       if (!exists) {
+        val archiveLogFolder = parameters.getOrElse(
+          HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, "archived")
         HoodieTableMetaClient.initTableType(sparkContext.hadoopConfiguration, path.get, tableType,
-          tblName.get, "archived", parameters(PAYLOAD_CLASS_OPT_KEY))
+          tblName.get, archiveLogFolder, parameters(PAYLOAD_CLASS_OPT_KEY))
```

Comment:
@hddong I'm accepting and will merge after you create another ticket for us to add to the release notes that if someone was actually overriding HOODIE_ARCHIVELOG_FOLDER_PROP_NAME that was not being honored before and now will be honored for those cases it can break (since this is the right thing to do)